### DISCUSSION TITLE
Using NOT_IN and IN inside the same token in Matcher

### DISCUSSION THREAD

**Discussion by inceatakan** [explosion/spaCy#10473] [Help: Coding & Implementations] [answered]
I am trying to write a pattern that distinguishes 'What do you think of the war?" from "What do you think would be a good case?' and 'What do you think that Mary likes?'. In other words, a pattern where the POS of the word after 'think' can be "PREP" but not "AUX"&"SCONJ". The pattern below is not helping:

[{'LOWER': {'IN': ['what','who','which']}, 'DEP': {"IN": ["nsubj","nsubjpass","pobj","dobj","det","attr",'dep','dat']}}, {'LOWER': 'else',"OP": "?"}, {'POS': 'AUX'},{'LEMMA': 'you'}, {'LEMMA': 'think'},{'POS':{"NOT_IN":['AUX',"DET","POSS","PROPN","ADJ"]},"POS":"PREP"}],

Is it possible to use NOT_IN and IN in a pattern for a single token that follows 'think' above?

Thanks in advance.

-Atakan

**Comment by adrianeboyd** [comment, marked as answer]
It's technically possible to use `NOT_IN` and `IN` for the same value in a token dict, but it doesn't really make sense as a pattern. I don't think there's any difference to just using `IN` with only the values that you want to match?

Be aware that if you use the same key twice in a dictionary definition, the last value is used:

```
assert {'POS':{"NOT_IN":['AUX',"DET","POSS","PROPN","ADJ"]},"POS":"PREP"} == {'POS': 'PREP'}
```

**Comment by inceatakan** [comment]
Thank you very much Adriane. That makes sense.
